domains/transfer-agreements/TransferAgreements.API/API.UnitTests/Controllers/TransferAgreementsControllerTests.cs

```diff
@@ -64,7 +64,7 @@ public async Task Create_ShouldCallRepositoryOnce()
             agreement.SenderId == Guid.Parse(subject) &&
             agreement.ActorId == atr &&
             agreement.StartDate == DateTimeOffset.FromUnixTimeSeconds(request.StartDate) &&
-            agreement.EndDate == DateTimeOffset.FromUnixTimeSeconds(request.EndDate) &&
+            agreement.EndDate == (request.EndDate.HasValue ? DateTimeOffset.FromUnixTimeSeconds(request.EndDate.Value) : null) &&
```

Comment:
Is this check relevant in the test? you could use the `!`instead `request.EndDate!.Value`